README.md

```diff
@@ -1,2 +1,155 @@
-# python-object-store-library
+# Python object store library for different platforms
+
 This is a python library for different object storages that have common methods.
+
+It is currently meant to support the following object storage platforms:
+
+- [MinIO](https://min.io/)
+- [Amazon S3](https://aws.amazon.com/s3/)
+- [Azure BLOB Container Storage](https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction)
+
+The library allows and adheres basic functionalities of object store platforms that are listed below:
+
+* connecting to particular object store
+* storing objects to buckets
+* retrieving object from buckets
+
+
+## Table of Contents
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+  - [Cleanup](#cleanup)
+  - [Usage](#usage)
+    - [Creating a specific object storage client](#creating-a-specific-object-storage-client)
+        - [MinIO](#minio)
+        - [AWS S3](#aws-s3)
+        - [Azure](#azure)
+    - [Reading from bucket](#reading-from-bucket)
+    - [Writing to bucket](#writing-to-bucket)
+
+## Prerequisites
+
+To properly build and use library, you have to install Python2 (with pip) and/or Python3 (with pip3). 
+
+## Installation
```

Comment:
The installation section should probably show how to install this package into a virtual environment since installing packages system-wide using `pip` just calling for troubles down the line.

Reply:
I agree and will modify this section.